sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -28,14 +28,24 @@ abstract sealed class SortDirection {
   def sql: String
 }
 
+// default null order is firt for asc
 case object Ascending extends SortDirection {
   override def sql: String = "ASC"
 }
 
+case object AscendingNullLast extends SortDirection {
```

Comment:
@hvanhovell One comment on this. To allow the 3rd parameter `nullOrdering` in the case class `SortOrder` definition, we need add `new` keyword in quite a few places where `SortOrder` object is constructed with default null ordering by the explicit constructor here. Otherwise it will not be able to compile. Is this OK?